classes/publication/Repository.php

```diff
@@ -77,7 +77,7 @@ public function add(Publication $publication): int
                 $submissionContext = Services::get('context')->get($submission->getData('contextId'));
             }
 
-            $supportedLocales = $submissionContext->getSupportedSubmissionLocales();
+            $supportedLocales = $publication->getLanguages($submissionContext->getSupportedSubmissionMetadataLocales());
```

Comment:
Shouldn't here be the same as in PKP-LIB, i.e.:
`$locales = $this->submission->getPublicationLanguages($this->context->getSupportedSubmissionMetadataLocales());`